lib/pry/commands/whereami.rb

```diff
@@ -44,11 +44,31 @@ class << self
       def setup
         if target.respond_to?(:source_location)
           file, @line = target.source_location
-          @file = expand_path(file)
         else
-          @file = expand_path(target.eval('__FILE__'))
+          file = target.eval('__FILE__')
           @line = target.eval('__LINE__')
         end
+
+        # If the current location is in a top-level script, then the filename
+        # returned by source_location or __FILE__ will be just the filename with
+        # no path: e.g. 'myscript.rb'. If the current working directory has been
+        # changed, then expand_path(file) will construct an incorrect path to
+        # the source. We can use __dir__ to fix this case.
+        if !file.nil? && !File.absolute_path?(file)
```

Comment:
As File.absolute_path? was added on Ruby 2.7. And we are still supporting older versions, is there any other way to make it compatible between versions?

Reply:
Yeah, that's disappointing. I guess the logic from `absolute_path?` could be back-ported inside of pry/whereami for earlier versions.

Reply:
I did my best to port the logic of `File.absolute_path?` to plain ruby inside of `whereami`. Can the tests for versions less than 2.7 be re-run with the latest code?

Also, since I copied 3 lines verbatim from a file from the ruby source repo to implement it, what's the best way to deal with the copyright issues? Ruby appears to be available under 2-clause BSD and pry looks to be under MIT, so I think they should be compatible, but I think we'd have to include the copyright info from ruby: